components/src/core/components/Input/Time/TimeInput.vue

```diff
@@ -79,59 +108,140 @@ export default defineComponent({
     },
   },
 
-  setup(props) {
+  setup(props, context) {
+    const datePickerOpen = ref<boolean>(false);
+    const amPmLabelFocus = ref<boolean>(false);
+
     const state = reactive({
-      open: false,
-      timeInput: props.modelValue,
+      time: '01:00',
+      am: true,
+      pickerInput: props.modelValue,
     });
 
-    return {
-      ...toRefs(state),
+    const openDropdown = () => {
+      datePickerOpen.value = true;
+      context.emit('timeselect:opened');
     };
-  },
 
-  methods: {
-    onFocusOut() {
-      if (this.open) {
-        this.$emit('update:modelValue', this.timeInput);
-        this.closeDropdown();
-      }
-    },
-    onTimeInput($event: Event) {
-      const input = ($event.target as HTMLInputElement).value;
-      this.timeInput = formatDate(parseDate(input, 'hh:mm a'), 'HH:mm');
-    },
-    toggleDropdown() {
-      if (!this.disabled) {
-        if (!this.open) {
-          this.$refs.oxdInput.$el.focus();
-          this.openDropdown();
+    const closeDropdown = () => {
+      datePickerOpen.value = false;
+      context.emit('timeselect:closed');
+    };
+
+    const toggleDropdown = () => {
+      if (!props.disabled) {
+        if (!datePickerOpen.value) {
+          openDropdown();
         } else {
-          this.closeDropdown();
+          closeDropdown();
         }
       }
-    },
-    openDropdown() {
-      this.open = true;
-      this.$emit('timeselect:opened');
-    },
-    closeDropdown() {
-      this.open = false;
-      this.$emit('timeselect:closed');
-    },
-  },
+    };
+
+    const onAmPmChange = (value: boolean) => {
+      state.am = value;
+    };
+
+    const toggleAmPm = () => {
+      state.am = !state.am;
+    };
+
+    const onTimeChange = (value: string) => {
+      state.time = value.trim();
+    };
+
+    const timePickerUpdate = () => {
+      context.emit('update:modelValue', state.pickerInput);
+    };
 
-  computed: {
-    timeIconClasses(): object {
+    watch(
+      () => props.modelValue,
+      () => {
+        if (props.modelValue) {
+          const time = parseDate(props.modelValue, 'HH:mm');
+          if (time) {
+            const formattedTime = formatDate(time, 'hh:mm');
+            if (formattedTime) {
+              state.time = formattedTime;
+              state.am = time.getHours() < 12;
+            }
+          }
+        }
+      },
+      {
+        immediate: true,
+      },
+    );
+
+    watch(
+      () => state,
```

Comment:
```suggestion
      () => state.time,
```
looks like you only need to watch for time.